src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java

```diff
@@ -133,62 +133,67 @@ public void encode(FacesContext context, DataTableRenderer renderer, DataTable t
     }
 
     public void filter(FacesContext context, DataTable table) {
```

Comment:
Check with @tandraschko and @melloware for this part, it's been recently modified and I didn't follow those changes

Reply:
actually minor change. filtering is processed only if filter values entered.
looks a lot because identation has changed.